File: ICM Technex/ICM09.py
t = int(input())

for _ in range(t):
    n = int(input())
    days = input()
    zeroes, ones, change = 0, 0, 0
    for i in range(n):
        if days[i] == '1':
            ones += 1
        elif days[i] == '0':
            zeroes += 1
        if ones < zeroes:
            change += 1
            ones += 1
            zeroes -= 1
    print(change)

File: ICM Technex/ICM09.cpp
#include<iostream>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while(t--)
    {
        int n;
        cin >> n;
        string days;
        cin >> days;
        int zeroes=0, ones=0, change=0;
        for(int i=0; i<n; i++)
        {
            if(days.at(i) == '1')
                ones++;
            else if(days.at(i) == '0')
                zeroes++;
            if(ones < zeroes)
            {
                change++;
                ones++;
                zeroes--;
            }
        }
        cout << change << endl;
    }
}